ibm/data_source_ibm_dl_port.go

```diff
@@ -0,0 +1,79 @@
+package ibm
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceIBMDirectLinkPort() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMDirectLinkPortRead,
+		Schema: map[string]*schema.Schema{
+			dlPortID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Port ID",
+			},
+			dlCount: {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Count of existing Direct Link gateways in this account on this port",
+			},
+			dlLabel: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Port Label",
+			},
+			dlLocationDisplayName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Port location long name",
+			},
+			dlLocationName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Port location long name",
+			},
+			dlProviderName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Port location long name",
+			},
+			dlSupportedLinkSpeeds: {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeInt,
+				},
+				Computed:    true,
+				Description: "Port's supported speeds in megabits per second",
+			},
+		},
+	}
+}
+
+func dataSourceIBMDirectLinkPortRead(d *schema.ResourceData, meta interface{}) error {
+
+	sess, err := meta.(ClientSession).DirectlinkV1API()
+	if err != nil {
+		return err
+	}
+
+	getPortsOptions := sess.NewGetPortOptions(d.Get(dlPortID).(string))
+	response, _, err := sess.GetPort(getPortsOptions)
+	if err != nil {
```

Comment:
When we try with an invalid port ID does this give any error?

Reply:
Yes. It gives, invalid ID error.